python/dgl/data/ppi.py

```diff
@@ -0,0 +1,121 @@
+"""PPI Dataset.
+(zhang hao): Used for inductive learning.
+"""
+
+import numpy as np
+from .utils import download, extract_archive, get_download_dir, _get_dgl_url
+from sklearn.preprocessing import StandardScaler
+from dgl import DGLGraph
+import networkx as nx
+from networkx.readwrite import json_graph
+import json
+
+_url = 'dataset/ppi.zip'
+
+class PPIDataset(object):
+    """Loads input data
+    ppi_G.json => the graph data used for training, test and validation as json format;
+    ppi-feats.npy => the feature vectors of nodes as numpy.ndarry object, it's shape is [n, v],
+    n is the number of nodes, v is the feature's dimension;
+    ppi-class_map.json => the classes of the input nodes, the format of it is {"1":[0, 1, 0, 1]},
+    "1" is the node index, [0, 1, 0, 1] is label;
+    ppi-graph_id.npy => the element in it indicates which graph the nodes belong to,
+    it is a one dimensional numpy.ndarray object and the length of it is equal the number of nodes,
+    it's like [1, 1, 2, 1,,,23];
+
+    """
+    name = 'ppi'
+    dir = get_download_dir()
+    zip_file_path = '{}/{}.zip'.format(dir, name)
+    download(_get_dgl_url(_url), path=zip_file_path)
+    extract_archive(zip_file_path,
+                    '{}/{}'.format(dir, name))
+    print('Loading G...')
+    with open('{}/ppi/ppi-G.json'.format(dir)) as jsonfile:
+        g_data = json.load(jsonfile)
+
+    with open('{}/ppi/ppi-class_map.json'.format(dir)) as jsonfile:
+        class_map = json.load(jsonfile)
+    features = np.load('{}/ppi/ppi-feats.npy'.format(dir))
+    label_list = []
+    for i in range(len(class_map)):
+        label_list.append(np.expand_dims(np.array(class_map[str(i)]), axis=0))
+    labels = np.concatenate(label_list)
+    graph = DGLGraph(nx.DiGraph(json_graph.node_link_graph(g_data)))
+    graph_id = np.load('{}/ppi/graph_id.npy'.format(dir))
```

Comment:
Put the above codes in `__init__`.

Reply:
If I put this part into `__init__`, it will load the data three times when create the train, test, validation and spend more time.

Reply:
My reply is accidentally cancelled. Have you received it?

Reply:
You are using class attributes in a wrong way. Class attributes will be evaluated with the class definition.
```python
class C:
  x = 1
  print('xxx')
# 'xxx' will be printed immediately after the above definition.
print(C.x)  # x become an attribute of the class not an instance
```

This means if user import `dgl.data.ppi`, the dataset will be loaded no matter what. User cannot control it. Even worse, if `from ppi import PPIDataset` is in `dgl.data.__init__`, then a user who wants to load other dataset will accidentally load PPIDataset.

There are many ways to fix it:
* Load the whole dataset and split the train/val/test sets outside the dataloader. I believer there should be clean ways to achieve that.
* Split the dataset beforehand.
* Add an `offset` argument in the `init` function to skip graphs before.

Reply:
I have add the `load_ppi` function to load the data, `PPIDataset` is used for preprocessing the dataset.

Reply:
That's not a good practice. Please have a look at [Pytorch's MNIST dataset](https://pytorch.org/docs/0.4.0/_modules/torchvision/datasets/mnist.html). They also have train and test set. See how they use one Dataset API to solve the problem.